packages/wrangler/src/__tests__/esbuild-define.test.ts

```diff
@@ -0,0 +1,118 @@
+import { mkdir, readFile, writeFile } from "node:fs/promises";
+import path from "node:path";
+import dedent from "ts-dedent";
+import { bundleWorker } from "../deployment-bundle/bundle";
+import { noopModuleCollector } from "../deployment-bundle/module-collection";
+import { mockConsoleMethods } from "./helpers/mock-console";
+import { runInTempDir } from "./helpers/run-in-tmp";
+import type { BundleOptions } from "../deployment-bundle/bundle";
+
+/*
+ * This file contains inline comments with the word "javascript"
+ * This signals to a compatible editor extension that the template string
+ * contents should be syntax-highlighted as JavaScript. One such extension
+ * is zjcompt.es6-string-javascript, but there are others.
+ */
+
+async function seedFs(files: Record<string, string>): Promise<void> {
+	for (const [location, contents] of Object.entries(files)) {
+		await mkdir(path.dirname(location), { recursive: true });
+		await writeFile(location, contents);
+	}
+}
+
+// Does bundleWorker respect the value of `defineNavigatorUserAgent`?
+describe("ESBuild defines are applied to the source code", () => {
+	runInTempDir();
+	mockConsoleMethods();
+
+	it("Replaces defined values when process.env.NODE_ENV is falsy", async () => {
+		const code = await getBundledWorker("");
+
+		expect(code).toContain(`userAgent = "Cloudflare-Workers";`);
+		expect(code).toContain(`g = global;`);
+		expect(code).toContain(`nodeEnv = "development";`);
+	});
+
+	it("Replaces defined values when process.env.NODE_ENV is undefined", async () => {
+		const code = await getBundledWorker(undefined);
+
+		expect(code).toContain(`userAgent = "Cloudflare-Workers";`);
+		expect(code).toContain(`g = global;`);
+		expect(code).toContain(`nodeEnv = "development";`);
```

Comment:
Where does this `development` value come from?

Reply:
Not sure... but this is not changed by this PR.
Is it ok if I create an issue to investigate and link it here to tackle in a follow up PR?

Reply:
Oh actually this comes from the `...define`:

```ts
		define: {
			...(defineNavigatorUserAgent
				? { "navigator.userAgent": `"Cloudflare-Workers"` }
				: {}),
			...(runtimeProcessEnv
				? { "process.env.NODE_ENV": `"${runtimeProcessEnv}"` }
				: {}),
			...(nodejsCompatMode === "legacy" ? { global: "globalThis" } : {}),
			...define,
		},
```

I need to investigate this more